src/meta/api/src/schema_api_impl.rs

```diff
@@ -810,7 +811,9 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
                             continue;
                         }
                         let db_meta = db_meta.unwrap();
-                        if is_drop_time_out_of_retention_time(&db_meta.drop_on, &now) {
+                        if req.filter.is_none()
+                            && is_drop_time_out_of_retention_time(&db_meta.drop_on, &now)
+                        {
```

Comment:
Why to skip a db_meta instance when filter is none?

It would be better adding some comment explaining the purpose.

Or explicitly examining the content of `req.filter`, e.g.: `if req.filter == Some(...) `